### ISSUE TITLE
Refreshing a deleted scoped token fails

### ISSUE THREAD

**Issue by bb-Ricardo** [jfrog/terraform-provider-artifactory#772]
**Describe the bug**
Creating a scope token works fine. But if the token has been deleted in the Artifactory UI then the refreshing during a `terraform plan` fails with:
```
│   22: resource "artifactory_scoped_token" "m2m_user_token" {
│ 
│ An unexpected error occurred while attempting to refresh resource state.
│ Please retry the operation or report this issue to the provider developers.
│ 
│ HTTP Error: 
│ 404 GET
│ [MASKED]/access/api/v1/tokens/5a6742e4-a7dc-4fb4-990f-dbdc8bdd8e76
│ {"errors":[{"code":"NOT_FOUND","detail":"Not Found","message":"HTTP 404 Not
│ Found"}]}
```

The issue stems from the handling of errors returned from resty for the request: https://github.com/jfrog/terraform-provider-artifactory/blob/34000181e7ad1af6a46c4e45a8bc5ff1464e0b3b/pkg/artifactory/resource/security/resource_artifactory_scoped_token.go#L410-L425

If Artifactory returns 404 resty will also return an error. The handling of the 404 needs to nested into the error handling block. And only if the status code != 404 the function `unableToRefreshResourceError` needs to be called.

I could offer to create a PR as well if desired.

**Requirements for and issue**
- [x] A description of the bug
- [ ] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). **If this is not supplied, this issue will likely be closed without any effort expended.**
- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version` 7.59.12
- [x] Your version of terraform: 1.4.3
- [x] Your version of terraform provider 8.3.1

**Expected behavior**
To recognise the missing token and offer to create a new token

**Comment by Valerianer** [comment]
I'm facing the same issue. 404 when running terraform plan or terraform apply, after the resource was deleted in the target-system. 

And I face the same issue with users as well.... is this already known? Shall a new issue be created for that?

**Comment by bb-Ricardo** [comment]
> I'm facing the same issue. 404 when running `terraform plan` or `terraform apply`, after the resource was deleted in the target-system. 
> 
> 
> 
> And I face the same issue with [users](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/user) as well.... is this already known? Shall a new issue be created for that?
> 
> 
> 
> terraform-version 2.5.1
> 
> terraform-artifactory-provider: 8.3.0

Quick glace at the code shows that users seemed to handled correctly. Do you have an example?

**Comment by danielmkn** [comment]
@Valerianer, if you have an issue with `user`, please create a new ticket with an example. Thanks.

**Comment by Valerianer** [comment]
Sure. Created with #775

**Comment by danielmkn** [comment]
Thanks for reporting! Fixed in v8.4.1
